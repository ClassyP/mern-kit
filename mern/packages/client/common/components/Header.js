import MoreIcon from 'material-ui/svg-icons/navigation/more-vert';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import AppBar from 'material-ui/AppBar';
import React, {Component} from 'react';

const Menu = (props) => {
  const iconButton = <IconButton><MoreIcon color='#FFF' /></IconButton>;
  const { links = {}} = props;

  return (
    <IconMenu
      iconButtonElement={iconButton}
      targetOrigin={{ horizontal: 'right', vertical: 'top' }}
      anchorOrigin={{ horizontal: 'right', vertical: 'top' }}>
      {links.logout ? <MenuItem primaryText='Logout' title='Logout' href='/auth/signout' /> : null}
    </IconMenu>
  );
};

export default class Header extends Component {
  render () {
    const {links} = this.props;

    return (
      <AppBar
        title='Mern-Kit'
        className='app__header'
        showMenuIconButton={false}
        iconElementRight={!links ? null : <Menu {...this.props} />} />
    );
  }
}
