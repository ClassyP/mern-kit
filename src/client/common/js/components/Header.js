import FlatButton from 'material-ui/FlatButton';
import AppBar from 'material-ui/AppBar';
import React, {Component} from 'react';

/* <header className='app__header'>
  <div className='inner'>
    <nav className='app__navigation'>
      <ul className='app__navigation-list'>
        <li className='app__navigation-list__item'>
          <FlatButton
            label='Logout'
            href='/auth/signout'
            labelStyle={labelStyle} />
        </li>
      </ul>
    </nav>
  </div>
</header> */

export default class Header extends Component {

  render () {
    const labelStyle = { textDecoration: 'underline', color: 'white' };
    const { links = {} } = this.props;

    return (
      <AppBar
        title='Mern-Kit'
        className='app__header'
        showMenuIconButton={false}>
        <nav className='app__navigation'>
          <ul className='app__navigation-list'>
            { !links.logout ? undefined :
              <li className='app__navigation-list__item'>
                <FlatButton
                  label='Logout'
                  href='/auth/signout'
                  labelStyle={labelStyle} />
              </li>
            }
          </ul>
        </nav>
      </AppBar>
    );
  }
}
