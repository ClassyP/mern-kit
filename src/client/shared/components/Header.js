import FlatButton from 'material-ui/FlatButton';
import React, {Component} from 'react';
import 'shared/css/header.scss';

export default class Header extends Component {

  render () {
    const labelStyle = { textDecoration: 'underline' };

    return (
      <header className='app__header'>
        <div className='inner'>
          <nav className='app__navigation'>
            <ul className='app__navigation-list'>
              <li className='app__navigation-list__item'>
                <FlatButton
                  label='Logout'
                  href='/auth/signout'
                  labelStyle={labelStyle} />
              </li>
            </ul>
          </nav>
        </div>
      </header>
    );
  }
}
