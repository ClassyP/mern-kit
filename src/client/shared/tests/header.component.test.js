import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Header from 'shared/components/Header';
import renderer from 'react-test-renderer';
import React from 'react';

/**
* Snapshot testing can be very useful if used correctly, this example just shows how to use
* it but this is not the most useful snapshot, sometimes it is better to call it multiple
* times inside a test and take snapshots of the component changing throughout some interactions,
* Ex. snapshot default state, snapshot after click/mouseover, snapshot after mouseleave
*/
describe('Header snapshot Tests', () => {

  test('Sample of snapshot testing, not very useful, but shows how to do it', () => {
    const header = renderer.create(
      <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
        <Header />
      </MuiThemeProvider>
    );
    const tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
