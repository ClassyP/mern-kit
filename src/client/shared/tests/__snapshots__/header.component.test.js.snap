// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Tests Should render a simple header with a link in it 1`] = `
<div
  className="app__header"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <h1
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    Mern-Kit
  </h1>
  <nav
    className="app__navigation"
  >
    <ul
      className="app__navigation-list"
    >
      <li
        className="app__navigation-list__item"
      >
        <a
          disabled={false}
          href="/auth/signout"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "minWidth": 88,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": "middle",
              "zIndex": 1,
            }
          }
          tabIndex={0}
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <span
              style={
                Object {
                  "color": "white",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "muiPrepared": true,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textDecoration": "underline",
                  "textTransform": "uppercase",
                  "verticalAlign": "middle",
                }
              }
            >
              Logout
            </span>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Header snapshot Tests Should render a simple header wrapped around an app bar 1`] = `
<div
  className="app__header"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <h1
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    Mern-Kit
  </h1>
  <nav
    className="app__navigation"
  >
    <ul
      className="app__navigation-list"
    />
  </nav>
</div>
`;
